manager/integration/tests/backupstore.py

```diff
@@ -0,0 +1,410 @@
+
+
+def backupstore_delete_volume_cfg_file(client, core_api, volume_name):  # NOQA
+    backupstore = backupstore_get_backup_target(client)
+
+    if is_backupTarget_s3(backupstore):
+        return minio_delete_volume_cfg_file(client,
+                                            core_api,
+                                            volume_name)
+
+    elif is_backupTarget_nfs(backupstore):
+        return nfs_delete_volume_cfg_file()
+
+
+# TODO: implement nfs_delete_volume_cfg_file
+def nfs_delete_volume_cfg_file():
+    raise NotImplementedError
+
+
+def minio_delete_volume_cfg_file(client, core_api, volume_name):
+    secret_name = backupstore_get_secret(client)
+    assert secret_name != ''
+
+    minio_api = minio_get_api_client(client, core_api, secret_name)
+    bucket_name = minio_get_backupstore_bucket_name(client)
+    minio_volume_cfg_file_path = minio_get_volume_cfg_file_path(volume_name)
+
+    try:
+        minio_api.remove_object(bucket_name, minio_volume_cfg_file_path)
+    except ResponseError as err:
+        print(err)
+
+
+def backupstore_create_dummy_in_progress_backup(client, core_api, volume_name):
+    backupstore = backupstore_get_backup_target(client)
+
+    if is_backupTarget_s3(backupstore):
+        return minio_create_dummy_in_progress_backup(client,
+                                                     core_api,
+                                                     volume_name,
+                                                     "backup-dummy")
+
+    elif is_backupTarget_nfs(backupstore):
+        return nfs_create_dummy_in_progress_backup()
+
+
+# TODO: implement nfs_create_dummy_in_progress_backup
+def nfs_create_dummy_in_progress_backup():
+    raise NotImplementedError
+
+
+def minio_create_dummy_in_progress_backup(client, core_api, volume_name, backup_name): # NOQA
+    dummy_backup_cfg_data = {"Name": "dummy_backup",
+                             "VolumeName": volume_name,
+                             "CreatedTime": ""}
+
+    backupstore_write_backup_cfg_file(client,
+                                      core_api,
+                                      volume_name,
+                                      backup_name,
+                                      dummy_backup_cfg_data)
+
+
+def backupstore_corrupt_backup_cfg_file(client, core_api, volume_name, backup_name): # NOQA
+    backupstore = backupstore_get_backup_target(client)
+
+    if is_backupTarget_s3(backupstore):
+        return minio_corrupt_backup_cfg_file(client,
+                                             core_api,
+                                             volume_name,
+                                             backup_name)
+
+    elif is_backupTarget_nfs(backupstore):
+        return nfs_corrupt_backup_cfg_file()
+
+
+# TODO: implement nfs_corrupt_backup_cfg_file
+def nfs_corrupt_backup_cfg_file():
+    raise NotImplementedError
+
+
+def minio_corrupt_backup_cfg_file(client, core_api, volume_name, backup_name): # NOQA
```

Comment:
Same here, you don't need the inbetween method. create the corrupt backup in backupstore_corrupt_backup_cfg_file and call the backupstore_write_backup_cfg_file method directly.

Reply:
fixed